python/cog/director/redis.py

```diff
@@ -73,3 +73,40 @@ def checker_() -> bool:
             return redis_key is not None and self.redis.exists(redis_key) > 0
 
         return checker_
+
+
+class RedisConsumerRotator:
+    def __init__(
+        self,
+        redis_configs: List[Tuple[str, str, str]],
```

Comment:
We could do that, yes. I'm not sure it would be simpler, though. We'd then need to parse argument lists and create `RedisConsumer`s and pass them through from somewhere else. I don't feel strongly about it, though, so happy to refactor if you thing it would be better the other way.

Reply:
I made this change in the end.